File: Advanced Methods of Programming/JavaClientServer/Persistence/src/main/java/Repo/RepoInscriere.java
package Repo;

import Model.Inscriere;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoInscriere implements IRepository<Integer, Inscriere> {
    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public RepoInscriere(Properties props) {
        logger.info("Initializing RepoInscriere with properties: {} ",props);
        this.dbUtils = new JdbcUtils(props);
    }

    public RepoInscriere() throws IOException {
        this.dbUtils = new JdbcUtils();
    }

    public void save(Inscriere inscriere) {
        logger.traceEntry("saving inscriere{}",inscriere);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into inscriere values (?,?,?)")){
            preStmt.setInt(1,inscriere.getIdParticipant());
            preStmt.setInt(2,inscriere.getIdProba());
            preStmt.setInt(3,inscriere.getId());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit();
    }

    @Override
    public int size() {
        logger.traceEntry();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select count(*) as SIZE from inscriere")) {
            try(ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    logger.traceExit(result.getInt("SIZE"));
                    return result.getInt("SIZE");
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        return 0;
    }

    @Override
    public Inscriere findOne(Integer id) {
        logger.traceEntry("finding inscriere with id {}",id);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from inscriere where idInscriere=?")) {
            preStmt.setInt(1,id);
            try(ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    Inscriere i= new Inscriere(result.getInt("idParticipant"),
                            result.getInt("idProba"));
                    logger.traceExit(i);
                    return i;
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit("No inscriere found with id{}",id);
        return null;
    }

    @Override
    public Iterable<Inscriere> findAll() {
        logger.traceEntry();
        List<Inscriere> lst = new ArrayList<>();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from inscriere")) {
            try(ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    lst.add(new Inscriere(result.getInt("idParticipant"),
                            result.getInt("idProba")));
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit(lst);
        return lst;
    }
}

File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Repo/RepoInscriere.cs
using Lab1C_MPP.Domain;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1C_MPP.Repo
{
    class RepoInscriere : IRepository<int, Inscriere>
    {
        //private static readonly ILog log = LogManager.GetLogger("SortingTaskDbRepository");

        IDictionary<String, string> props;
        public RepoInscriere(IDictionary<String, string> props)
        {
            //log.Info("Creating RepoAngajat");
            this.props = props;
        }

        public IEnumerable<Inscriere> FindAll()
        {
            IDbConnection con = DBUtils.getConnection(props);
            IList<Inscriere> inscrieri = new List<Inscriere>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select idParticipant,idProba,idInscriere from inscriere" +
                    " where idInscriere=@idInscriere";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idParticipant = dataR.GetInt32(0);
                        int idProba = dataR.GetInt32(1);
                        int idInscriere = dataR.GetInt32(2);
                        Inscriere inscriere = new Inscriere(idProba,idParticipant);
                        inscrieri.Add(inscriere);
                    }
                }
            }

            return inscrieri;
        }

        public Inscriere FindOne(int Id)
        {
            //log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select idParticipant,idProba,idInscriere from inscriere" +
                    " where idInscriere=@idInscriere";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@idInscriere";
                paramId.Value = Id;
                comm.Parameters.Add(paramId);
                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        int idParticipant = dataR.GetInt32(0);
                        int idProba = dataR.GetInt32(1);
                        int idInscriere = dataR.GetInt32(2);
                        Inscriere inscriere = new Inscriere(idProba, idParticipant);
                        //log.InfoFormat("Exiting findOne with value {0}", task);
                        return inscriere;
                    }
                }
            }
            //log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public void Save(Inscriere inscriere)
        {
            var con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into inscriere  values (@idParticipant, @idProba, @idInscriere)";
                var paramIdPa = comm.CreateParameter();
                paramIdPa.ParameterName = "@idParticipant";
                paramIdPa.Value = inscriere.GetIdParticipant();
                comm.Parameters.Add(paramIdPa);

                var paramIdPr = comm.CreateParameter();
                paramIdPr.ParameterName = "@idProba";
                paramIdPr.Value = inscriere.GetIdProba();
                comm.Parameters.Add(paramIdPr);

                var paramIdI = comm.CreateParameter();
                paramIdI.ParameterName = "@idInscriere";
                paramIdI.Value = inscriere.GetId();
                comm.Parameters.Add(paramIdI);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                    throw new RepositoryException("Nu s-a adaugat nicio inscriere!");
            }
        }

        public int Size()
        {
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select count(*) from inscriere";
                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        //log.InfoFormat("Exiting findOne with value {0}", task);
                        return dataR.GetInt32(0);
                    }
                }
            }
            //log.InfoFormat("Exiting findOne with value {0}", null);
            return 0;
        }
    }
}
